src/test/java/com/alphawallet/attestation/demo/VerifyAttestation.sol

```diff
@@ -0,0 +1,793 @@
+        
+        if (rhs[0] == 0 && rhs[1] == 0) { return false; } //early revert to avoid spending more gas
+
+        //Add result of riddle.multiply(c) to point
+        (x, y) = extractXYFromPoint(pok.tPoint);
+        uint256[2] memory tPointCoords = [x, y];
+        rhs = ecAdd(rhs, tPointCoords);
+        
+        return ecEquals(lhs, rhs);
+    }
+
+    function ecMul(uint256 s, uint256 x, uint256 y) public view
+        returns (uint256[2] memory retP)
+    {
+        bool success;
+        // With a public key (x, y), this computes p = scalar * (x, y).
+        uint256[3] memory i = [x, y, s];
+        
+        assembly 
+        {
+            // call ecmul precompile
+            // inputs are: x, y, scalar
+            success := staticcall (not(0), 0x07, i, 0x60, retP, 0x40)
+        }
+        
+        if (!success)
+        {
+            retP[0] = 0;
+            retP[1] = 0;
+        }
+    }
+  
+    function ecInv(uint256[2] memory point) private view 
+        returns (uint256[2] memory invPoint)
+    {
+        invPoint[0] = point[0];
+        int256 n = int256(fieldSize) - int256(point[1]);
+        n = n % int256(fieldSize);
+        if (n < 0) { n += int256(fieldSize); }
+        invPoint[1] = uint256(n);
+    }
+    
+    function ecAdd(uint256[2] memory p1, uint256[2] memory p2) public view
+        returns (uint256[2] memory retP)
+    {
+        bool success;
+        uint256[4] memory i = [p1[0], p1[1], p2[0], p2[1]];
+        
+        assembly 
+        {
+            // call ecadd precompile
+            // inputs are: x1, y1, x2, y2
+            success := staticcall (not(0), 0x06, i, 0x80, retP, 0x40)
+        }
+        
+        if (!success)
+        {
+            retP[0] = 0;
+            retP[1] = 0;
+        }
+    }
+    
+    function extractXYFromPoint(bytes memory data) public pure returns (uint256 x, uint256 y)
+    {
+        bytes32 xBytes;
+        bytes32 yBytes;
+        assembly 
+        {
+            xBytes := mload(add(data, 0x21)) //copy from 33rd byte because first 32 bytes are array length, then 1st byte of data is the 0x04; 
+            yBytes := mload(add(data, 0x41)) //65th byte as x value is 32 bytes.
+        }
+        
+        x = uint256(xBytes);
+        y = uint256(yBytes);
+    }
+    
+    function extractXYFromPoint2(bytes memory data) public pure returns (uint256 x, uint256 y)
+    {
+        bytes32 xBytes;
+        bytes32 yBytes;
+        assembly 
+        {
+            xBytes := mload(add(data, 0x21)) //copy from 33rd byte because first 32 bytes are array length, then 1st byte of data is the 0x04; 
+            yBytes := mload(add(data, 0x41)) //65th byte as x value is 32 bytes.
+        }
+        
+        x = uint256(xBytes);
+        y = uint256(yBytes);
+    }
+
+    function mapToCurveMultiplier(bytes memory input) public pure returns(uint256 res)
+    {
+        bytes memory nextInput = input;
+        do
+        {
```

Comment:
I will make the change in the rejection-sampling branch to ensure that `uint256(idHash) >> curveOrderBitShift` is less than curve order. Thus the loop will not be needed, but only a check that `uint256(idHash) >> curveOrderBitShift` is in fact less than curve order. If not the client has been malicious and the request should fail.